bgpd/bgp_route.c

```diff
@@ -529,6 +529,9 @@ static uint32_t bgp_med_value(struct attr *attr, struct bgp *bgp)
 void bgp_path_info_path_with_addpath_rx_str(struct bgp_path_info *pi, char *buf,
 					    size_t buf_len)
 {
+	if (!pi)
```

Comment:
I've just looked in the code and can't spot any _real_ call where we can pass NULL here. Did you have something specific here? I'm concerned if we won't mask some problems in the future with this.

Reply:
We came across this coverity warning: `Access to field 'host' results in a dereference of a null pointer (loaded from field 'peer')`, that's why included this check for pi.
We can always remove it. 
Please let us know.

Reply:
Coverity doesn't say anything about `pi`, it says that `host` may be NULL. I didn't check whether it is true, but the check you added doesn't protect from this at all.

Reply:
Ok. Will drop this commit.